R/aggregate_results.R

```diff
@@ -0,0 +1,240 @@
+        sum(.data$plan_carsten, na.rm = TRUE),
+      technology_exposure = .data$asset_portfolio_value *
+        sum(.data$plan_carsten, na.rm = TRUE),
+      technology_value_change = .data$technology_exposure * .data$VaR_technology / 100
+    ) %>%
+    dplyr::ungroup()
+
+  return(data)
+}
+
+aggregate_value_change_to_sector_level <- function(data) {
+  data <- data %>%
+    dplyr::group_by(
+      .data$investor_name, .data$portfolio_name, .data$ald_sector,
+      .data$scenario_geography
+    ) %>%
+    dplyr::mutate(
+      VaR_sector = sum(.data$VaR_technology * .data$plan_carsten, na.rm = TRUE) /
+        sum(.data$plan_carsten, na.rm = TRUE),
+      sector_exposure = .data$asset_portfolio_value *
+        sum(.data$plan_carsten, na.rm = TRUE),
+      sector_value_change = .data$sector_exposure * .data$VaR_sector / 100
+    ) %>%
+    dplyr::ungroup()
+
+  return(data)
+}
+
+aggregate_value_change_to_analysed_sectors_and_portfolio <- function(data) {
+  data <- data %>%
+    dplyr::group_by(
+      .data$investor_name, .data$portfolio_name, .data$scenario_geography
+    ) %>%
+    dplyr::mutate(
+      VaR_analysed_sectors = sum(.data$VaR_tech_company * .data$plan_carsten, na.rm = TRUE) /
+        sum(.data$plan_carsten, na.rm = TRUE),
+      analysed_sectors_exposure = .data$asset_portfolio_value *
+        sum(.data$plan_carsten, na.rm = TRUE),
+      analysed_sectors_value_change = .data$analysed_sectors_exposure *
+        .data$VaR_analysed_sectors / 100,
+      portfolio_aum = .data$asset_portfolio_value,
+      # setting portfolio_value_change = analysed_sectors_value_change will
+      # underestimate overall impact on portfolio as there can of course be
+      # impacts on companies in the portfolio that operate in other sectors
+      portfolio_value_change = .data$analysed_sectors_value_change,
+      portfolio_value_change_perc = sum(.data$VaR_tech_company * .data$tech_company_exposure, na.rm = TRUE) /
+        .data$portfolio_aum
+    ) %>%
+    dplyr::ungroup()
+
+  return(data)
+}
+
+prepare_exposure_for_credit_risk_format <- function(data) {
+  # distinct in order to remove unused technology level information, the
+  # trivial year and the unneeded financial info
+  data <- data %>%
+    dplyr::distinct(
+      .data$investor_name, .data$portfolio_name, .data$company_name,
+      .data$ald_sector, .data$scenario_geography, .data$plan_sec_carsten
+    )
+
+  return(data)
+}
+
+join_credit_exposure_on_pd_changes <- function(data,
+                                               exposure) {
+  data <- data %>%
+    dplyr::inner_join(
+      exposure,
+      by = c(
+        "investor_name", "portfolio_name", "company_name",
+        "scenario_geography", "ald_sector"
+      )
+    )
+
+  return(data)
+}
+
+aggregate_pd_change_to_sector_level <- function(data,
+                                                horizon = c("annual", "overall")) {
```

Comment:
maybe add a check that horizon is of length 1, this design pattern could be misunderstoof for a default.